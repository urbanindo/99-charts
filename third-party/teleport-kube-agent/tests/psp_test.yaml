suite: PodSecurityPolicy
templates:
  - psp.yaml
tests:
  - it: creates a PodSecurityPolicy when enabled in values
    set:
      podSecurityPolicy:
        enabled: true
    asserts:
      - hasDocuments:
          count: 3
      - documentIndex: 0
        isKind:
          of: PodSecurityPolicy
      - documentIndex: 1
        isKind:
          of: Role
      - documentIndex: 2
        isKind:
          of: RoleBinding
      - matchSnapshot: {}

  - it: sets PodSecurityPolicy labels when specified
    values:
      - ../.lint/extra-labels.yaml
    asserts:
      - equal:
          path: metadata.labels.app\.kubernetes\.io/name
          value: teleport-kube-agent
      - equal:
          path: metadata.labels.resource
          value: podsecuritypolicy
      - matchSnapshot: {}
