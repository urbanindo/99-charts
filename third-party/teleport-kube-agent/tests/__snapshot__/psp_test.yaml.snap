creates a PodSecurityPolicy when enabled in values:
  1: |
    apiVersion: policy/v1beta1
    kind: PodSecurityPolicy
    metadata:
      annotations:
        seccomp.security.alpha.kubernetes.io/allowedProfileNames: docker/default,runtime/default
        seccomp.security.alpha.kubernetes.io/defaultProfileName: runtime/default
      name: RELEASE-NAME
    spec:
      allowPrivilegeEscalation: false
      fsGroup:
        ranges:
        - max: 65535
          min: 1
        rule: MustRunAs
      hostIPC: false
      hostNetwork: false
      hostPID: false
      privileged: false
      readOnlyRootFilesystem: true
      requiredDropCapabilities:
      - ALL
      runAsUser:
        rule: MustRunAsNonRoot
      seLinux:
        rule: RunAsAny
      supplementalGroups:
        ranges:
        - max: 65535
          min: 1
        rule: MustRunAs
      volumes:
      - '*'
sets PodSecurityPolicy labels when specified:
  1: |
    apiVersion: policy/v1beta1
    kind: PodSecurityPolicy
    metadata:
      annotations:
        seccomp.security.alpha.kubernetes.io/allowedProfileNames: docker/default,runtime/default
        seccomp.security.alpha.kubernetes.io/defaultProfileName: runtime/default
      labels:
        app.kubernetes.io/name: teleport-kube-agent
        resource: podsecuritypolicy
      name: RELEASE-NAME
    spec:
      allowPrivilegeEscalation: false
      fsGroup:
        ranges:
        - max: 65535
          min: 1
        rule: MustRunAs
      hostIPC: false
      hostNetwork: false
      hostPID: false
      privileged: false
      readOnlyRootFilesystem: true
      requiredDropCapabilities:
      - ALL
      runAsUser:
        rule: MustRunAsNonRoot
      seLinux:
        rule: RunAsAny
      supplementalGroups:
        ranges:
        - max: 65535
          min: 1
        rule: MustRunAs
      volumes:
      - '*'
